writers_rollingfilewriter.go

```diff
@@ -138,7 +140,7 @@ var compressionTypes = map[rollingArchiveType]compressionType{
 			if err != nil {
 				return err
 			}
-			return createGzip(archiveName, tar)
+			return createGzip(archiveName, bytes.NewReader(tar))
```

Comment:
The idea was to have the rolling file writer stream the archived files directly when the destination format matches the source format and stream the archived files to disk to get the FileInfo and compute the respective header before streaming it to the destination.

I actually implemented something like this here (still WIP): https://github.com/jonsyu1/seelog/tree/jyu/archive

The advantage here is we avoid streaming it multiple times when the formats match (the 99% case). I also don't see any external dependencies in seelog, so I'm trying to avoid introducing them: https://github.com/jonsyu1/seelog/search?q=github.com&type=Code&utf8=%E2%9C%93